File: exercise/kotlin/week06-exercise3/src/main/kotlin/order/OrderProcessor.kt
package order

class OrderProcessor {
    fun processOrders(orders: List<Order>) {
        for (item in orders) {
            if (item.status == Order.UNPROCESSED) {
                if (item.items > 5) {
                    // apply bulk discount
                    item.total *= 0.9
                }
                item.status = Order.PROCESSED
            }
        }
    }
}

File: exercise/c#/week06-legacy-code/OrderProcessor/OrderProcessor.cs
namespace OrderProcessor
{
    public class OrderProcessor
    {
        public void ProcessOrders(List<Order> orders)
        {
            foreach (var item in orders)
            {
                if (item.Status == Order.Unprocessed)
                {
                    if (item.Items > 5)
                    {
                        // apply bulk discount
                        item.Total *= 0.9;
                    }

                    item.Status = Order.Processed;
                }
            }
        }
    }
}